File: C++/Fact2.cpp
#include <iostream>
using namespace std;

long fact(int n) {
	long f=1;
	while( n>=1 ) { 
		f = n*f;
		n--;
	}
	return f;
}


int main() {	
	int n;
	cout << "Input number: ";
	cin >> n;
	
	cout << fact(n) << "\n";
	return 0;
}

File: Node/Fact2.js
// ====== 共通の入出力機能（変更しない）======
import { output, input } from "./io.js";
// ==========================================

function fact(n) {
    let f = 1;
    while (n >= 1) {
        f = n * f;
        n--;
    }
    return f;
}


function main() {
    const n = parseInt(input("Input number: "));

    output(fact(n) + "\n");
}

main();
